python/ray/serve/schema.py

```diff
@@ -116,17 +127,50 @@ def valid_encoding_format(cls, v):
 
         return v
 
+    @validator("access_log")
+    def valida_acccess_log(cls, v):
+
+        if v not in list(AcessLoggingType):
+            raise ValueError(
+                f"Got '{v}' for access_log. access_log must be one "
+                f"of {set(AcessLoggingType)}."
+            )
+
+        return v
+
     @validator("log_level")
     def valid_log_level(cls, v):
         if isinstance(v, int):
-            return v
+            if v not in logging._levelToName:
+                raise ValueError(
+                    f'Got "{v}" for log_level. log_level must be one of '
+                    f"{list(logging._levelToName.keys())}."
+                )
+            return logging._levelToName[v]
 
         if v not in logging._nameToLevel:
             raise ValueError(
                 f'Got "{v}" for log_level. log_level must be one of '
                 f"{list(logging._nameToLevel.keys())}."
             )
-        return logging._nameToLevel[v]
+        return v
+
+    def _should_enable_stream_logging(self):
+        return self.access_log in [AcessLoggingType.ALL, AcessLoggingType.STREAM_ONLY]
+
+    def _should_enable_file_logging(self):
+        return self.access_log in [AcessLoggingType.ALL, AcessLoggingType.FILE_ONLY]
+
+    @property
+    def version(self):
```

Comment:
Can we rename this and make it private? `version` seems a bit out of context (maybe just name it `_hash`?) if you don't know it's being used in `DeploymentVersion`, and this doesn't need to be exposed to the user.

Reply:
write __eq__ func